File: app/src/main/java/com/imangazalievm/bubbble/data/repository/UsersRepository.java
package com.imangazalievm.bubbble.data.repository;

import com.imangazalievm.bubbble.data.network.DribbbleApiService;
import com.imangazalievm.bubbble.domain.models.User;
import com.imangazalievm.bubbble.domain.repository.IUsersRepository;

import javax.inject.Inject;

import io.reactivex.Single;

public class UsersRepository implements IUsersRepository {

    private DribbbleApiService dribbbleApiService;

    @Inject
    public UsersRepository(DribbbleApiService dribbbleApiService) {
        this.dribbbleApiService = dribbbleApiService;
    }

    @Override
    public Single<User> getUser(long userId) {
        return dribbbleApiService.getUser(userId);
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/data/users/UsersRepository.kt
package com.imangazalievm.bubbble.data.users

import com.imangazalievm.bubbble.data.global.network.DribbbleApi
import com.imangazalievm.bubbble.domain.global.models.User
import io.reactivex.Single
import javax.inject.Inject

class UsersRepository @Inject constructor(
    private val dribbbleApi: DribbbleApi
) {

    suspend fun getUser(userId: Long): User {
        return dribbbleApi.getUser(userId)
    }

}